### DISCUSSION TITLE
LAB 02.04 Task 3: Run the card trick

### DISCUSSION THREAD

**Discussion by BustamJos3** [rramosp/ai4eng.v1#144] [02 - PYTHON]
Hola, agradecería que me brindaran ayuda en saber qué está pasando:
Definiendo la función así:
![image](https://user-images.githubusercontent.com/72939041/174663239-a39fc7a9-e3ab-47bc-982c-5bd27027e137.png)
Obtengo lo siguiente:
1. 
![image](https://user-images.githubusercontent.com/72939041/174663490-8f85a3e2-8f11-4c72-abe2-d37627b83303.png)
2.
![image](https://user-images.githubusercontent.com/72939041/174663431-df3d1034-ece5-4826-81a8-69f9874a6af5.png)
3. 
![image](https://user-images.githubusercontent.com/72939041/174663597-bdcee2e1-03a0-475e-9b47-8e2629c610ab.png)
Y el grader me dice esto:
![image](https://user-images.githubusercontent.com/72939041/174663700-d4128549-8aa6-485c-8ff9-90a2211b2fd9.png)

No sé por qué no me está dando la posición correcta. Agradezco cualquier ayuda  @Felipe-RA

**Comment by BustamJos3** [comment]
@raulramosudea @rramosp

**Comment by Felipe-RA** [comment]
Saludos @BustamJos3 

Gracias por el tag, se me había perdido tu post.

Primero, recordarte que es preferible utilizar bloques de código a capturas de pantalla para incluir tu código (**Lineamientos para formular preguntas eficientes** #145) .

Esto lo puedes lograr con la siguiente sintaxis: 

\```

\```

Ahora, en cuanto a tu problema:

Partiendo de que tus funciones `make_heap()` y `collect_heap()` estén correctas; el problema que identifico es la forma en la que las llamas para correr el truco de las cartas, específicamente cuando buscas la **picked card** `n`:

¿Debes de buscar a `n` cada vez que haces los heap y los colectas? ¿O sólo al final?

Revisa cuándo deberías de utilizar `.index(n)`

Estoy pendiente cualquier cosa

Saludos
